microsoft/omnichannel-chat-widget
# Attachment File Scan

## **Thank you for your contribution. Before submitting this PR, please include:**

### Id of the task, bug, story or other reference.

### Description
- Currently, attachment files with `in progress` or `malicious` states can be downloaded

## Solution Proposed
- Any attachment files not with `passed` state won't be able to be downloaded. 
- The `File.content` of the attachment in `malicious` state be will removed. 
- Attachment files with `in progress` state would be added to the file scanning, its state would be periodically verified until we get a definite state (`passed` or `malicious`), then it would be handled accordingly

### Acceptance criteria
- Uploading file attachments should go through file scan
- File currently being scanned or is malicious should not be downloadable

## Test cases and evidence
1. Upload single image attachment
   - Attachment should be ignored since `scan.status` only exists for file attachments and not image attachments
1. Upload single file attachment which takes time to retrieve the `scan.status` (.pdf, .pptx)
   - Attachment should go from `in progress` to `passed`
1. Upload malicious file attachment
   - It should be blocked
1. Upload image + file attachment
   - Multiple attachments should be uploaded accordingly
1. Upload multiple files attachment
   - Multiple attachments should be uploaded accordingly

### Sanity Tests
-   [ ] You have tested all changes in Popout mode
-   [ ] You have tested all changes in cross browsers i.e Edge, Chrome, Firefox, Safari and mobile devices(iOS and Android)
-   [ ] Your changes are included in the [CHANGELOG](../CHANGE_LOG.md)


## A11y
- [ ] You have run the [Automated Accessibility Check](https://accessibilityinsights.io/docs/en/windows/getstarted/automatedchecks) and have no reported issues

__*Please provide justification if any of the validations has been skipped.*__
